modules/apps/27-interchain-accounts/host/keeper/handshake.go

```diff
@@ -53,15 +53,6 @@ func (k Keeper) OnChanOpenTry(
 		if channel.IsOpen() {
 			return "", errorsmod.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s is already OPEN", activeChannelID, portID)
 		}
-
-		appVersion, found := k.GetAppVersion(ctx, portID, activeChannelID)
```

Comment:
just for my own reasoning. The previous metadata has:
version, controller/host connection id's, address, encoding, tx type

Only the host connection id and address cannot change? The host connection id check above in `ValidateHostMetadata`, but is the address? What happens if the controller reopens a channel modifying the interchain acc address to be different than what it is. It looks like it'll be overwritten later on. I feel like we should document these cases or make them more explicit

Reply:
If I'm understanding correctly, the address is looked up with a key built from host connection id and counterparty port id, right? And counterparty port ID needs to proven by core before getting here and if host connection id cannot change then it should be all fine.

Reply:
It's just a bit odd to me because the relayer can pass in:
metadata.Address = invalid-address
and our code won't error
it'll return 
metadata.Address = valid-address

just feels a bit odd if a middleware relies on the value before the callback

Reply:
That's right. Moreover, the removed code (`IsPreviousMetadataEqual`) never actually checked if the addresses were the same... It only checked other fields

Reply:
I like being a bit defensive/explicit in these cases:
```go
// loosen restrictions, but keep explicit checks
// the host connectionID and address must remain the same between reopening
appVersion, found := k.GetAppVersion(ctx, portID, activeChannelID)
if !found {
	panic(fmt.Errorf("active channel mapping set for %s, but channel does not exist in channel store", activeChannelID))
}
		
previousMetadata, err := MetadataFromVersion(previousVersion)
if err != nil {
     return err // or panic
 }
 
if previousMetadata.HostConnectionId != metadata.HostConnectionId {
   return err
}
 
if previousMetadata.Address != metadata.Address {
     return err
}   
 

```

but I don't feel strongly

Reply:
I was going to suggest the same about at least explicitly checking the address. I didn't think of the middleware point you brought up, good catch. 

I'd be in favour of adding the extra defensive checks because why not! But also not really gonna block on it

Reply:
ah okay, thanks for the context. Should we add a comment in case someone has the same question as me? 

maybe something like:
```go
// if a channel is being reopened, we allow the controller to propose new fields
// which are not exactly the same as the previous. The provided address will 
// be overwritten with the correct one before the metadata is returned
```

Reply:
Added the comment

Reply:
Should we check at least that version is `ics27-1`?

Reply:
This is still being checked [here](https://github.com/cosmos/ibc-go/blob/198b525a0b870df2c3198e56151266bd334daf3e/modules/apps/27-interchain-accounts/host/keeper/handshake.go#L42) @crodriguezvega